psyneulink/core/llvm/codegen.py

```diff
@@ -298,9 +298,19 @@ def gen_composition_exec(ctx, composition, *, tags:frozenset):
 
         # Generate a while not 'end condition' loop
         builder.position_at_end(loop_condition)
+        is_finished_flags = [None] * len(composition.nodes)
+        for idx, node in enumerate(composition.nodes):
+            #FIXME: This skips nested nodes
+            from psyneulink import Composition
```

Comment:
Do not import `Composition` here. Check for the presence of `is_finished_flag` in state instead.

Reply:
Is the `is_finished_flag` attribute necessarily guaranteed to only exist on `Compositions`?

Reply:
I'm not sure what the question implies. Are compositions considered to be always finished?
what is the expected behaviour when the flag is not present?

It'd also be better to construct a node indexed dict instead of a list. that way you can skip nodes that don't have the flag available and get an error if a scheduling condition tries to use such a node.